config/interpolate_funcs.go

```diff
@@ -704,6 +706,25 @@ func interpolationFuncBase64Sha256() ast.Function {
 	}
 }
 
+func interpolationFuncRandomHex() ast.Function {
+	return ast.Function{
+		ArgTypes:   []ast.Type{ast.TypeInt},
+		ReturnType: ast.TypeString,
+		Callback: func(args []interface{}) (interface{}, error) {
+			n := args[0].(int)
+			b := make([]byte, n)
+
+			_, err := rand.Read(b)
+			if err != nil {
+				return "", fmt.Errorf("failed to read random bytes: %v", err)
+			}
+
+			f := fmt.Sprintf("%%0%dx", n)
+			return fmt.Sprintf(f, b)[0:n], nil
+		},
```

Comment:
Alternative implementation, but I disliked the conversion to/from `big.Int`: https://gist.github.com/justincampbell/17c6e2f21ae25307335541ea33cefe85

It may be worthwhile to separate out a `generateRandomString()` function which takes a dictionary and length.